As a user, I want to override the classloader in the default configuration

So, because I have different classloaders in my app I'm getting this error:
<details>
<summary>Stacktrace</summary>

```
Caused by: java.lang.ExceptionInInitializerError
	at com.avast.grpc.jwt.keycloak.server.KeycloakJwtServerInterceptor.fromConfig(KeycloakJwtServerInterceptor.java:17) ~[?:?]
	at org.camunda.community.zeebe.keycloak.interceptor.ZeebeKeycloakInterceptor.<init>(ZeebeKeycloakInterceptor.java:18) ~[?:?]
	at org.camunda.community.zeebe.keycloak.interceptor.ZeebeKeycloakInterceptor.<init>(ZeebeKeycloakInterceptor.java:14) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:?]
	at io.camunda.zeebe.util.ReflectUtil.newInstance(ReflectUtil.java:18) ~[zeebe-util-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.interceptors.impl.InterceptorRepository.instantiate(InterceptorRepository.java:110) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.interceptors.impl.InterceptorRepository.lambda$instantiate$0(InterceptorRepository.java:56) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.util.HashMap$EntrySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at io.camunda.zeebe.gateway.Gateway.applyInterceptors(Gateway.java:204) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.Gateway.start(Gateway.java:123) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.StandaloneGateway.run(StandaloneGateway.java:88) ~[camunda-cloud-zeebe-1.2.12.jar:1.2.12]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:777) ~[spring-boot-2.6.6.jar:2.6.6]
	... 5 more
Caused by: com.typesafe.config.ConfigException$Missing: system properties: No configuration setting found for key 'keycloakDefaults'
	at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:157) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:175) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:189) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:194) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.getObject(SimpleConfig.java:269) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.getConfig(SimpleConfig.java:275) ~[?:?]
	at com.typesafe.config.impl.SimpleConfig.getConfig(SimpleConfig.java:42) ~[?:?]
	at com.avast.grpc.jwt.keycloak.KeycloakFactory.<clinit>(KeycloakFactory.java:10) ~[?:?]
	at com.avast.grpc.jwt.keycloak.server.KeycloakJwtServerInterceptor.fromConfig(KeycloakJwtServerInterceptor.java:17) ~[?:?]
	at org.camunda.community.zeebe.keycloak.interceptor.ZeebeKeycloakInterceptor.<init>(ZeebeKeycloakInterceptor.java:18) ~[?:?]
	at org.camunda.community.zeebe.keycloak.interceptor.ZeebeKeycloakInterceptor.<init>(ZeebeKeycloakInterceptor.java:14) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:?]
	at io.camunda.zeebe.util.ReflectUtil.newInstance(ReflectUtil.java:18) ~[zeebe-util-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.interceptors.impl.InterceptorRepository.instantiate(InterceptorRepository.java:110) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.interceptors.impl.InterceptorRepository.lambda$instantiate$0(InterceptorRepository.java:56) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
	at java.util.HashMap$EntrySpliterator.forEachRemaining(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
	at io.camunda.zeebe.gateway.Gateway.applyInterceptors(Gateway.java:204) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.Gateway.start(Gateway.java:123) ~[zeebe-gateway-1.2.12.jar:1.2.12]
	at io.camunda.zeebe.gateway.StandaloneGateway.run(StandaloneGateway.java:88) ~[camunda-cloud-zeebe-1.2.12.jar:1.2.12]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:777) ~[spring-boot-2.6.6.jar:2.6.6]
	... 5 more
```
</details>

And the root cause of it is: `Caused by: com.typesafe.config.ConfigException$Missing: system properties: No configuration setting found for key 'keycloakDefaults'`. I think that this happening because in the default mode `Thread.currentThread().getContextClassLoader()` is used. So I need somehow to pass another classloader to the default settings.

--- a/keycloak/src/main/java/com/avast/grpc/jwt/keycloak/KeycloakFactory.java
+++ b/keycloak/src/main/java/com/avast/grpc/jwt/keycloak/KeycloakFactory.java
@@ -6,11 +6,15 @@ import org.keycloak.admin.client.Keycloak;
 import org.keycloak.admin.client.KeycloakBuilder;
 
 public final class KeycloakFactory {
-  public static Config DefaultConfig =
-      ConfigFactory.defaultReference().getConfig("keycloakDefaults");
+
+  public static final String KEYCLOAK_DEFAULTS_CONFIG_NAME = "keycloakDefaults";
 
   public static Keycloak fromConfig(Config config) {
-    Config fc = config.withFallback(DefaultConfig);
+    return fromConfig(config, Thread.currentThread().getContextClassLoader());
+  }
+
+  public static Keycloak fromConfig(Config config, ClassLoader contextClassLoader) {
+    Config fc = config.withFallback(getDefaultConfig(contextClassLoader));
     return KeycloakBuilder.builder()
         .clientId(fc.getString("clientId"))
         .clientSecret(fc.getString("clientSecret"))
@@ -20,6 +24,10 @@ public final class KeycloakFactory {
         .realm(fc.getString("realm"))
         .serverUrl(fc.getString("serverUrl"))
         .build();
+  }
+
+  public static Config getDefaultConfig(final ClassLoader classLoader) {
+    return ConfigFactory.defaultReference(classLoader).getConfig(KEYCLOAK_DEFAULTS_CONFIG_NAME);
   }
 
   private KeycloakFactory() {}

--- a/keycloak/src/main/java/com/avast/grpc/jwt/keycloak/server/KeycloakJwtServerInterceptor.java
+++ b/keycloak/src/main/java/com/avast/grpc/jwt/keycloak/server/KeycloakJwtServerInterceptor.java
@@ -1,6 +1,6 @@
 package com.avast.grpc.jwt.keycloak.server;
 
-import static com.avast.grpc.jwt.keycloak.KeycloakFactory.DefaultConfig;
+import static com.avast.grpc.jwt.keycloak.KeycloakFactory.getDefaultConfig;
 
 import com.avast.grpc.jwt.server.JwtServerInterceptor;
 import com.avast.grpc.jwt.server.JwtTokenParser;
@@ -14,7 +14,12 @@ public class KeycloakJwtServerInterceptor extends JwtServerInterceptor<AccessTok
   }
 
   public static KeycloakJwtServerInterceptor fromConfig(Config config) {
-    Config fc = config.withFallback(DefaultConfig);
+    return fromConfig(config, Thread.currentThread().getContextClassLoader());
+  }
+
+  public static KeycloakJwtServerInterceptor fromConfig(
+      Config config, ClassLoader contextClassLoader) {
+    Config fc = config.withFallback(getDefaultConfig(contextClassLoader));
     KeycloakPublicKeyProvider publicKeyProvider =
         new DefaultKeycloakPublicKeyProvider(
             fc.getString("serverUrl"),

